src/components/CityBlock/CityBlock.jsx

```diff
@@ -0,0 +1,115 @@
+import React, { useEffect, useState } from "react";
+
+import { getWeatherByCity } from "../../service/weather/weather.service";
+
+import style from "./city-block.module.scss";
+import commonStyles from "../../common.module.scss";
+
+const CityBlock = ({
+  cityItem,
+  setModalProps,
+  isAuthorized,
+  handlePinCity,
+}) => {
+  const [currWeather, setCurrWeather] = useState(null);
+  const [isPinned, setIsPinned] = useState(false);
```

Comment:
why isPinned stored here? It look like. a property of the city, and should be handled by parent, since you have to call parent anyway